### PR TITLE
Templatize and fix builders

### PR DESCRIPTION
**Pull request by bavery22**
This is a pretty big change. I figured out why all the redhat builders were breaking and didnt want to change each one. In short, this
1) makes a template for the builder dockerfiles and has build_container fill it in.
2) moves runbitbake.py from /home/yoctouser to /usr/local/bin

### CODE REVIEW

File: build_container.sh [hunk 1]
```diff
@@ -51,6 +51,9 @@ cd -
 # can modify the Dockerfile to use whatever REPO is in the environment.
 TAG=$DISTRO_TO_BUILD-builder
 dockerdir=`find -name $TAG`
+# use the builder template to populate the distro specific Dockerfile
+cp dockerfiles/templates/Dockerfile.builder $dockerdir/Dockerfile
+sed -i "s/DISTRO_TO_BUILD/$DISTRO_TO_BUILD/g" $dockerdir/Dockerfile
```
Comment (incandescant) [inline build_container.sh]: Why the sed & cp, why not
`sed "s/DISTRO_TO_BUILD/$DISTRO_TO_BUILD/g" dockerfiles/templates/Dockerfile.builder > $dockerdir/Dockerfile`

Probably not worth changing the patch just for this.

File: dockerfiles/centos/centos-7/centos-7-builder/README [hunk 2]
```diff
@@ -0,0 +1,2 @@
+The Dockerfile for this directory is autogenerated based on dockerfiles/templates/Dockerfile.builder
+Do not place an actual Dockerfile here.
```
Comment (incandescant) [inline dockerfiles/centos/centos-7/centos-7-builder/README]: We should add the -builder directories to the gitignore, that will discourage people from putting Dockerfiles there and help avoid the generated files being added to the repo.

File: .gitignore [hunk 3]
```diff
@@ -1,3 +1,4 @@
+dockerfiles/*/*/*-builder/Dockerfile
```
Comment (incandescant) [inline .gitignore]: Ignore my last…

**Comment by bavery22** [comment]
based on incandescant review, merging...

File: .gitignore (hunk 3 above)
Comment (bavery22) [inline .gitignore]: based on incandescant review, merging...
--

**Pull request merged**